### ISSUE TITLE
Question about evaluating the model on roxford dataset

### ISSUE THREAD

**Issue by feymanpriv** [ducha-aiki/affnet#21]
I have tested the released model on roxford dataset, but the final result is not good as follows:

  roxford5k: mAP E: 67.17, M: 53.37, H: 30.37
  roxford5k: mP@k[ 1 5 10] E: [95.59 72.06 55.88], M: [97.14 81.43 62.86 ], H:[78.57 31.43 14.29]
My scheme is:

1. Use hessian detector and affnet to extract 3000 features and hardnet++ to extract descriptors which is 3000*128.
2. Use Ransac to find good matches between each query and reference and use the sum of matchMask as score.
3. Use revisitop to evaluate the result.

Hope to get some suggestions.

**Comment by ducha-aiki** [comment]
Hi,

Well, that is expected, as your pipeline is completely different from one used in paper:

1. Detection was used with constant threshold 5.33 on Hessian response instead of just 3000k per image
2. Matching was done by BoW scheme + Hamming embedding 
3. Top-1000 matches very spatially verifying (similar to what you have done)
4. For verified matched, query expansion was done. 

![image](https://user-images.githubusercontent.com/4803565/54348239-e7ad6180-4648-11e9-85c0-7143bafefb3e.png)


[13] is https://hal.inria.fr/hal-00971267/PDF/tolias_lqe14.pdf

**Comment by feymanpriv** [comment]
Thanks, but i wonder if just use the simple strategy without using the methods you have mentioned, what the result will be.

**Comment by ducha-aiki** [comment]
Your simple strategy is not scalable, unfortunately. 
But if you want to improve results:

1) make sure that you do not use orientation detector, but gravity-vector (do_ori = False)

2) opencv ransac is not good. Try this one https://github.com/ducha-aiki/zeromqransac or this https://github.com/cr333/usac-cmake

**Comment by feymanpriv** [comment]
@ducha-aiki Thank you very much! I will try recently.

**Comment by feymanpriv** [comment]
@ducha-aiki When the feature is extracted as what you said(do_ori = False, th=5.33), I have improved 10% of mAP. And i am trying to realize your BoW scheme + Hamming embedding + SV process. 
1. But in the experiments, i found the feature with affnet(AffNet=AffNetPix) or without affnet(AffNet=None) just have a little difference. I wonder if i do something wrong. As i checked, the output of LAFs (x, y, a, b, c) and descriptors are different. Need some other operations?
2. If 'do_ori = False', then the keypoints extracted don't include orientation information and OriNet is no need to use, is it right?

**Comment by ducha-aiki** [comment]
1. It is possible for some images: if not viewpoint change, AffNet might produce circular shape (aka no affine)
2. yes

@ym547559398 I have found my old private repo with naive implementationretrieval similar to you described in first post, but with query expansion and SV.  It reads the extracted features in specific format and then does benchmark on new oxform5k/paris6k 
It gives following results:

```
oxford5knew mAP E: 82.1, M: 70.26, H: 51.84
oxford5knew mP@k[1, 5, 10, 20, 100] 
E: [94.12 76.47 72.06 61.76 47.06],
M: [95.71 88.57 74.29 55.71 24.29],
H: [90.   65.71 44.29 25.71 14.29]
```

If you want, I can share with you that private repo. But keep it mind, that it is completely undocumented, no guarantees and probably I would have no time to answer the questions about those code

**Comment by feymanpriv** [comment]
@ducha-aiki That's great! I will compare your private code with mine carefully. And i just have some questions with your QE  and SV process, if you share, i can learn it myself. Also, i have implemented  a simple pipeline of bow+he+sv process during these days. Thank you for patient reply and wonderful work again! 'AffNet' is beyond this and look forward to your further research.

**Comment by feymanpriv** [comment]
@ducha-aiki And you can send to 547559398@qq.com or hangman@bupt.edu.cn. Thank you~

**Comment by ducha-aiki** [comment]
I have sent you github invite. If I remember correctly, you can run test_534h.py file

**Comment by feymanpriv** [comment]
@ducha-aiki OK，thanks

**Comment by feymanpriv** [comment]
@ducha-aiki Hi, i have tested the code this week. And I got a final result of 67.77 with qe and sv. The base is similar to my process, although 67.77 is much higher than base with your QE process, it still has a gap to get your result. I have checked the process of extracting feature again, the parameters are same with yours, i wonder if there is something i still ignored. Hope to get your help again!
